packages/react/src/query/prepareSwap.ts

```diff
@@ -0,0 +1,38 @@
+import type { Config } from "wagmi";
+import type { QueryOptions } from "@tanstack/react-query";
+import { filterQueryOptions } from "./utils.js";
+import {
+  prepareSwap,
+  type PrepareSwapParameters,
+  type PrepareSwapReturnType,
+} from "@alchemy/wagmi-core";
+
+// TODO(jh): may want to use `UnionExactPartial<PrepareSwapParameters>` here. test it first.
+export type PrepareSwapOptions = PrepareSwapParameters;
+
+function prepareSwapQueryKey(options: PrepareSwapOptions) {
+  const { connector: _connector, ...rest } = options;
+  return ["prepareSwap", filterQueryOptions(rest)] as const;
+}
+
+export type PrepareSwapQueryKey = ReturnType<typeof prepareSwapQueryKey>;
+
+export function prepareSwapQueryOptions(
```

Comment:
i separated these from the actual hook files to follow wagmi's pattern. technically these should probably even live in `wagmi-core` so that they can be used in non-react frameworks, but i'm hoping to punt that for now b/c i've been having an extremely rough time getting that to work w/ our current monorepo setup. (i expect it will be a lot easier once we get rid of v4, update the examples, and switch to pnpm.)